app/controllers/courses_controller.rb

```diff
@@ -2,7 +2,7 @@ class CoursesController < ApplicationController
   before_filter :load_course, except: [:index]
 
   def index
-    @courses = Course.paginate(page: params[:page], per_page: DEFAULT_PER_PAGE).decorate
+    @courses = Course.includes(:institute).paginate(page: params[:page], per_page: DEFAULT_PER_PAGE).decorate
```

Comment:
It's Rails best practice to try and keep your controllers "thin", ie very simple.

```suggestion
    @courses = Course.for_index(page: params[:page], per_page: DEFAULT_PER_PAGE).decorate
```

the put the logic in Course class 
```ruby
class Course < ApplicationRecord
  def self.for_index(page:, per_page: )
    includes(:institute).paginate(page: page, per_page: per_page)
  end
end
```
**More Background**
Exposing a performance optimisation in the controller distracts from the core of what needs to happen in the controller. That said, in the "wild" (real RTO project) this could still be acceptable. It's a slippery slope as complexity does not happen all at once, but one step at a time.

Reply:
I think this is a matter of personal opinion and whatever guidelines the project specifies. Personally I think the `for_index` method is specifically related to the view or presentation side of things and so should reside in the controller rather than the model. Perhaps if not in the controller you could put it as a class method in the presenter but I don't really see the benefit in that.

Reply:
agreed, this would change from team to team if this is acceptable. 
I do personal think it's better to keep active-record specifics in the models and decorators and helpers to not know anything about them. 
At this point it most likely be a good idea for the team to decide how they want to proceed and what is acceptable but happy to accept this on the basis that it's still simple enough to understand.